performance-es.md

```diff
@@ -57,3 +57,117 @@ La optimización prematura también puede afectarte al atarte a ciertas decision
 En la gran mayoría de casos, el tamaño y velocidad del programa no es el problema. La optimización más fácil es no hacerla. La segunda alternativa más fácil es simplemente comprar mejor hardware.
 
 Cuando hayas decidido cambiar tu programa, sigue leyendo.
+
+### Modificar los datos
+
+Modificar los datos significa agregar o alterar la representación de los datos
+que estas procesando. Desde el punto de vista del rendimiento, algunos de
+estos acabaran cambiando la complejidad O() asociada a diferentes aspectos de
+la estructura de datos.
+
+Ideas para mejorar tu estructura de datos:
+
+* Campos adicionales
+
+  El clásico ejemplo de esto es almacenar la longitud de una lista enlazada en un
+  campo en el nodo raíz. Mantenerla actualizada lleva un poco más de trabajo,
+  pero consultar la longitud se convierte en un simple acceso a un campo en vez de una
+  búsqueda transversal con complejidad O(n). Tu estructura de datos puede presentar una
+  mejora similar: un poco de mantenimiento en algunas operaciones a cambio de
+  mejorar el rendimiento en un caso de uso común.
+
+  De manera similar, almacenar punteros a nodos frecuentemente utilizados en vez
+  de realizar búsquedas adicionales. Esto cubre cosas como el link "hacia atrás"
+  en una lista doblemente enlazada para hacer que la eliminación de nodos tenga una complejidad O(1).
+  Algunas listas de salto guardan un "puntero de búsqueda", donde almacenan un
+  puntero a donde recientemente estuviste en tu estructura de datos, bajo la
+  suposición de que es un buen punto de partida para la siguiente operación.
+
+* Índices de búsqueda adicionales
+
+  La mayoría de las estructuras de datos están diseñadas para un único tipo de
+  consulta. Si necesitas dos tipos de consultas, disponer de una "vista"
+  adicional a tus datos puede ser una gran mejora. Por ejemplo, un set de
+  structs puede tener un ID primario (integer) que usas para buscar en un
+  slice, pero a veces necesitas buscar por un ID secundario (string). En lugar
+  de iterar sobre el slice, puedes mejorar tu estructura de datos con un mapa de
+  cadena a ID o directamente al struct en cuestión.
```

Comment:
Something like this was brought up in slack. Can we use the actual `string` word instead of translating it? Just like how we use `slice` across language?

Reply:
Oh, totally. That one slipped past by me from before discussing that